drake/CMakeLists.txt

```diff
@@ -42,13 +47,15 @@ function(add_library_with_exports)
   cmake_parse_arguments(parsed_args "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
 
   add_library(${parsed_args_LIB_NAME} SHARED ${parsed_args_SOURCE_FILES})
+  set(exports_abs_path ${PROJECT_BINARY_DIR}/exports/drake/${parsed_args_LIB_NAME}_export.h)
   # work around https://cmake.org/Bug/view.php?id=12645 (fixed in 2.8.8):
-  file(RELATIVE_PATH exports_rel_path ${CMAKE_CURRENT_BINARY_DIR}/ ${PROJECT_BINARY_DIR}/exports/${parsed_args_LIB_NAME}_export.h)
+  file(RELATIVE_PATH exports_rel_path ${CMAKE_CURRENT_BINARY_DIR}/ ${exports_abs_path})
   generate_export_header(${parsed_args_LIB_NAME} EXPORT_FILE_NAME ${exports_rel_path})
-  # TODO: enable when 2.8.11 is required
-#  target_include_directories(${parsed_args_LIB_NAME} PUBLIC ${PROJECT_BINARY_DIR}/exports)
+  install(FILES ${exports_abs_path} DESTINATION include/drake/)
 endfunction()
-include_directories(${PROJECT_BINARY_DIR}/exports)
+include_directories(${PROJECT_BINARY_DIR}/exports/)
+include_directories(${PROJECT_SOURCE_DIR}/..)
```

Comment:
i think we have to be very careful here.  if I include, e.g., `drake/test/testUtil.h` (the first example below), then it's actually going to get the previously installed version of that file, not the file in the source directory.  Because `build/include` is already put into the include_directories from pods.cmake (if I remember correctly).  

i think you want this line to be 
```
include_directories(BEFORE ${PROJECT_SOURCE_DIR}/..)
```
(see https://cmake.org/cmake/help/v3.0/command/include_directories.html )

even then, it's a bit of a slippery slope.  but i guess that's probably the right thing.